pkg/controller/controller.go

```diff
@@ -334,6 +343,36 @@ func ttls() []string {
 	return []string{"15m", "30m", "1h", "2h", "4h"}
 }
 
-func categories() []string {
-	return []string{"explanation"}
+func (c *Controller) categories() []string {
+	validators := c.p.Validators()
+	categories := make([]string, 0, len(validators))
+	for v := range validators {
+		categories = append(categories, v)
+	}
+	categories = append(categories, "explanation")
+	return categories
+}
+
+// categoryPairs returns a list of Pairs for rendering.
+func (c *Controller) categoryPairs() []Pair {
+	pairs := make([]Pair, 0, len(c.categories()))
+	for _, category := range c.categories() {
+		pairs = append(pairs, Pair{
+			Key:  category,
+			Text: getText(category),
+		})
+	}
+	return pairs
+}
+
+// getText returns the corresponding text for rendering.
+func getText(category string) string {
+	switch category {
+	case "explanation":
+		return "Explanation"
+	case "jira":
+		return "Jira"
+	default:
+		return "UNKNOWN"
+	}
```

Comment:
This feels like it's going to be fragile and break over time. Can the plugin's expose this themselves instead?

Reply:
For now, It's only the plugin name. We can probably do https://pkg.go.dev/strings#ToTitle.

Reply:
+1 on passing back the display data from plugin.

I have been playing with different functions under `strings` package, but don't seem a way to make things camel case.  Note that @qiaoweiz is working on passing back human-friendly category and hint from validator plugin, as a result I simply add a getText function here to stay compatible with existing code.

Reply:
You need to use cases, not strings. I added it in #303.

Reply:
#303 is merged and hopefully it can simplify some logic in this PR

Reply:
Yep yep, the logic of current PR is simplified a lot now. PTAL